static/js/containers/PostPage.js

```diff
@@ -0,0 +1,89 @@
+// @flow
+import React from 'react';
+import { connect } from 'react-redux';
+import R from 'ramda';
+
+import Card from '../components/Card';
+import Loading, { anyProcessing, allLoaded } from '../components/Loading';
+import ChannelBreadcrumbs from '../components/ChannelBreadcrumbs';
+import PostDisplay from '../components/PostDisplay';
+
+import { actions } from '../actions';
+
+import type { Dispatch } from 'redux';
+import type { Match } from 'react-router';
+
+class PostPage extends React.Component {
+  props: {
+    match:            Match,
+    dispatch:         Dispatch,
+    posts:            Object,
+    channels:         Object,
+  };
+
+  getMatchParams = () => {
+    const { match: { params }} = this.props;
+    return [params.postID, params.channelName];
+  }
+
+  updateRequirements = () => {
+    const { dispatch, channels, posts } = this.props;
+    const [postID, channelName] = this.getMatchParams();
+
+    if (R.isNil(posts.data.get(postID))) {
+      dispatch(actions.posts.get(postID));
+    }
+    if (R.isNil(channels.data.get(channelName))) {
+      dispatch(actions.channels.get(channelName));
+    }
+  }
+
+  componentWillMount() {
+    this.updateRequirements();
+  }
+
+  componentDidUpdate() {
+    const { channels, posts } = this.props;
+    let restStates = [ channels, posts ];
+
+    if (!anyProcessing(restStates) || allLoaded(restStates)) {
+      this.updateRequirements();
+    }
+  }
+
+  renderContents(postID, posts, channelName, channels) {
+    const post = posts.get(postID);
+    const channel = channels.get(channelName);
+
+    if (R.isNil(channel) || R.isNil(post)) {
```

Comment:
Is this check necessary? It seems like `renderContents` only gets called if all requests have finished